docs/generating-tests.rst

```diff
@@ -0,0 +1,417 @@
+``"network" : ALL``.
+
+Filling the Test
+----------------
+
+The test filler file is not for consumption.  The filler file needs to be filled
+into a test.  ``testeth`` has the ability to compute the post-state from the test filler,
+and produce the test.  The advantage of the filled test is that it can catch any post-state
+difference between clients.
+
+First, if you created a new subdirectory for the filler, you need to edit the source
+of ``cpp-ethereum`` so that ``testeth`` recognizes the new subdirectory.  The file to 
+edit is `StateTests.cpp`_, which lists the names of the subdirectories scanned for 
+GeneralStateTest filters.
+
+.. _`StateTests.cpp`: https://github.com/ethereum/aleth/blob/master/test/tools/jsontests/StateTests.cpp
+
+
+After building ``testeth``, you are ready to fill the test.
+
+.. code:: bash
+
+   ETHEREUM_TEST_PATH="<LOCAL_PATH_TO_ETH_TESTS>" test/testeth -t GeneralStateTests/stExample2 -- --filltests --checkstate
+
+
+where the environmental variable ``ETHEREUM_TEST_PATH`` should point to the directory 
+where ``tests`` repository is checked out.  ``stExample2`` should be replaced with 
+the name of the subdirectory you are working on.  ``--filltests`` option tells ``testeth``
+to fill tests.  ``--checkstate`` tells ``testeth`` to check the final states against
+the ``expect`` fields.
+
+Depending on your shell, there are various ways to set up ``ETHEREUM_TEST_PATH``
+environment variable.  For example, adding ``export ETHEREUM_TEST_PATH=/path/to/tests``
+to ``~/.bashrc`` might work for ``bash`` users.
+
+``testeth`` with ``--filltests`` fills every test filler it finds. The command might
+modify existing test cases. After running ``testeth`` with ``--filltests``\ , try running
+``git status`` in the ``tests`` directory. If ``git status`` indicates changes in 
+unexpected files, that is an indication that the behavior of ``cpp-ethereum`` changed unexpectedly.
+
+.. note::
+   If ``testeth`` is looking for tests in the ``../../test/jsontests`` directory, 
+   you have probably not specified the ``--testpath`` option.
+
+
+Trying the Filled Test
+----------------------
+
+Trying the Filled Test Locally
+++++++++++++++++++++++++++++++
+
+For trying the filled test, in ``aleth/build`` directory, run
+
+.. code:: bash
+
+   ETHEREUM_TEST_PATH="../../tests" test/testeth -t GeneralStateTests/stExample2
+
+
+Trying the Filled Test in Travis CI
++++++++++++++++++++++++++++++++++++
+
+Moreover, for trying the filled test in ``Travis CI`` for ``ethereum/aleth``, 
+the new test cases need to exist in a branch in ``ethereum/tests``. For this,
+ask somebody with a push permission to ``ethereum/tests``.
+
+After that, enter ``aleth/test/jsontests`` directory, and checkout the branch in
+``ethereum/tests``.  Then go back to ``aleth`` directory and perform 
+``git add test/jsontests`` followed by ``git commit``.
+
+When you file this commit as a pull-request to ``ethereum/aleth``, Travis CI 
+should try the newly filled tests.
```

Comment:
If I understand this correctly, the aim is the get the Aleth Travis CI build to run the new tests with Aleth to check they pass.  To do that a PR is submitted to Aleth that updates the git submodule for `ethereum/tests` to point to a branch with the new tests.  It may be worth giving an overview like that to explain what we're trying to do before launching into how.  I had to read it a few times to understand it without the context.

It should also say how you get in touch with someone who can create the branch in `ethereum/tests` for you.

Even better would be to setup a CI job so that this test is just automatically run for each PR submitted to `ethereum/tests`. Having to get someone to create a branch in `ethereum/tests` for you is a pretty big barrier.

Reply:
Will try to do a summary of this in the documentation. Will think about the implications and how the process is currently handled and could be eventually improved when I work on the doc section update.

Reply:
aleth is a legacy actually. the logic you described is aleth specific. the actuall test repo should not really care about aleth. I've been working on migrating the test logic from aleth into a separate test tool. one we release the hardfork the priority would be to make a production version of that tool that would be capable of generating/running the tests from test repo using any compatible client.

Reply:
Clarified the whole context by doing additions from both of your comments in the section.